posts/2023-09-29-open-liberty-on-aws-ecs-fargate.adoc

```diff
@@ -0,0 +1,684 @@
+
+=== Key Considerations when using Amazon ECS with Fargate
+Amazon ECS with Fargate does provide arguably the most configurable Container as a Service offering from any of the major cloud providers. However, It has also been a challenge to configure all the necessary pieces given a very significant number of assumptions that AWS makes about our container that become apparent during wizard based deployments such as use of non-default ports.
+
+If it had not been for non-default ports, a significant number of the permissions that were required as a non-admin in the end would not have been needed as the ECS Wizard does cover basic creation of target groups and ALBs, however once you are off the golden path or needing to do something above what the wizard is capable of
+
+Compared to AWS Lambda, it is easier to lift an existing containerized application into Amazon ECS given that there is no need to use the Amazon SDK or Runtime API to send and receive work and for web applications it is a much better fit.
+
+There are a number of benefits over other CaaS offerings
+
+* More control over the deployment as you do have access to the various resources that you create
+* More flexibility in the configuration at multiple levels
+* More granular access control such that it is possible to create better divisions of responsibility
+* Can define related microservices in a single task definition and scale all together. The scaling is at a ratio of 1:1, so you cannot scale one container independently of another. Competitors typically only allow the deployment of a single container per definition.
+* Can scale on CPU or Memory metrics as well as HTTP requests, which might better fit an application's  profile.
+
+While Amazon ECS with Fargate running services does provide a highly scalable serverless architecture it does have some limitations compared to its competitors such as IBM Cloud Code Engine(ICCE) or Azure Container Apps(ACS) for running Web Applications:
+
+* No scale to 0. To be able to handle requests you need at least 1 running instance. Both ICCE and ACS support scale from 0 for HTTP workloads.
+* Networking management. Even basic networking needs the deployer to configure and manage their network resources.
+* Harder to manage given mix of Amazon ECS and Amazon EC2 resources needed for a single application.
+* Harder to get typical real world examples working quickly
+* Scaling can take some time to start to scale out as it requires 3 readings above the set threshold before it will act.
+* No auto-generation of trusted certificate for new HTTPS ALBs, especially if using the `amazonaws` domain
+* Integration with other AWS services is not well documented with the focus being on AWS Lambda instead of Amazon ECS
+* Large number of permissions required for a single person to deploy all resources needed by an application as a non-admin.
+
+== Appendices
```

Comment:
We shouldn’t have “Appendices” in a blog post. That info should be mentioned when needed and/or linked to for detail.